core/runtime/src/main/java/io/quarkus/runtime/util/IsWindows.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.runtime.util;
+
+import java.util.function.BooleanSupplier;
+
+import io.smallrye.common.os.OS;
+
+/**
+ * Platform specific annotations,
+ * e.g. @TargetClass(className = "sun.java2d.windows.WindowsFlags", onlyWith = IsWindows.class)
+ */
+public class IsWindows implements BooleanSupplier {
```

Comment:
@geoand Would this be an idiomatic way in Quarkus? I need to drive some substitutions OS specific :-)

Reply:
@zakkak Why would that be though? I will be adding IsMacOS soon too. It's a public utility consumable in any runtime package... Its usage should have no side effects. 

Perhaps if I enrich it with version detection, it's gonna be more messy, trying to pull containers to read GraalVM version etc...